code/magento-os/lightna-theme/Magento_Theme/web/js/lightna.js

```diff
@@ -0,0 +1 @@
sest("li");await o.post(this.removeFromCartUrl,t,{onSuccess:this.onProductRemove.bind(this,s)})}onProductRemove(e){e.classList.add(this.classes.fade),setTimeout((()=>{this.update.bind(this)()}),this.minActionDuration)}},new class{menuContainer;itemsTopLevel;itemsWithChildren;delay;mobileBreakpoint=768;mouseEnterTimeout=null;mouseEnterBlocked=!1;classNames={activeItem:"active",activeOverlay:"overlay-active",hoverItem:"hover",preventClick:"prevent-click"};selectors={itemTopLevelTitle:".level-1__title",itemsWithChildren:".level-1.has-children .level-1__title",itemsTopLevel:".level-1",hasChildren:".has-children",menuContainer:".main-navigation"};constructor(e={}){this.delay=c()?0:e.delay||150,this.menuContainer=n(this.selectors.menuContainer),this.itemsWithChildren=r(this.selectors.itemsWithChildren),this.itemsTopLevel=r(this.selectors.itemsTopLevel),this.menuContainer.classList.remove("no-js"),this.bindEvents()}bindEvents(){document.addEventListener("mouseenter",this.preventMenuActivation.bind(this),{once:!0}),this.itemsWithChildren.forEach((e=>{e.addEventListener("click",this.itemWithChildrenClick.bind(this,e))})),this.itemsTopLevel.forEach((e=>{e.addEventListener("mouseenter",this.itemTopLevelMouseEnter.bind(this,e)),e.addEventListener("mouseleave",this.itemTopLevelMouseLeave.bind(this,e))}))}preventMenuActivation(){c()||this.itemsTopLevel.forEach((e=>{e.matches(":hover")&&(this.mouseEnterBlocked=!0)}))}itemWithChildrenClick(e,t){if(window.innerWidth<this.mobileBreakpoint)return this.toggleMobileSubmenu(e),void t.preventDefault();e.matches(`.${this.classNames.preventClick}`)&&(t.preventDefault(),e.classList.remove(this.classNames.preventClick))}itemTopLevelMouseEnter(e){this.mouseEnterBlocked||(c()&&n(this.selectors.itemTopLevelTitle,e).classList.add(this.classNames.preventClick),this.mouseEnterTimeout=setTimeout((()=>{const t=n(`.${this.classNames.hoverItem}`,this.menuContainer);t?.classList.remove(this.classNames.hoverItem),e.classList.add(this.classNames.hoverItem),this.toggleOverlay(!0,e.matches(this.selectors.hasChildren))}),this.delay))}itemTopLevelMouseLeave(e){this.mouseEnterBlocked=!1,clearTimeout(this.mouseEnterTimeout);const t=e.previousElementSibling?.matches(":hover"),s=e.nextElementSibling?.matches(":hover");t||s||setTimeout((()=>{const e=n(`.${this.classNames.hoverItem}`,this.menuContainer);e&&(e.classList.remove(this.classNames.hoverItem),this.toggleOverlay(!1))}),this.delay)}toggleMobileSubmenu(e){e.parentNode.classList.toggle(this.classNames.activeItem)}toggleOverlay(e,t=!1){document.body.classList.toggle(this.classNames.activeOverlay,e&&t)}},new class{mobileOverlayTriggers;constructor(){this.mobileOverlayTriggers=r(".cjs-toggle-overlay"),this.bindEvents()}bindEvents(){this.mobileOverlayTriggers.forEach((e=>{e.addEventListener("click",(()=>{this.toggleMobileOverlay(e.dataset.overlay)}))}))}toggleMobileOverlay(e){n(`[data-overlay-id="${e}"]`).classList.toggle("active"),document.body.classList.toggle("mobile-overlay-active")}}})();
```

Comment:
it won't be ok to have this way as lightna js changes every time by its modules. we need to include it directly from lightna

Reply:
but then we will load more of js...

Reply:
like in 2 times?

Reply:
likely
i'd say it depends how semi-lightna will be used - if we consider that all we have there designed to be used in magento and we don't have anything extra, smth that can break/conflict with magento js - we can load all of it